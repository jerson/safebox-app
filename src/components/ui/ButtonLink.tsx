import * as React from 'react';
import {
  GestureResponderEvent,
  StyleProp,
  StyleSheet,
  TextStyle,
  ViewStyle
} from 'react-native';
import Touchable, { TouchableProps } from './Touchable';
import Colors from '../../modules/constants/Colors';
import Font from '../../modules/resources/Font';
import Text from './Text';

const styles = StyleSheet.create({
  container: {
    alignSelf: 'center',
    padding: 6
  } as ViewStyle,
  strong: {
    ...Font({ weight: 'Bold' })
  } as TextStyle,
  title: {
    color: Colors.grey5,
    fontSize: 13
  } as TextStyle,
  accent: {
    color: Colors.primary
  } as TextStyle
});

export interface ButtonLinkProps extends TouchableProps {
  isLoading?: boolean;
  isDisabled?: boolean;
  title?: string;
  strong?: boolean;
  accent?: boolean;
  titleStyle?: StyleProp<TextStyle>;
}

export interface ButtonLinkState {}

function ButtonLink(props: ButtonLinkProps) {
  const {
    onPress,
    isLoading,
    isDisabled,
    title,
    strong,
    accent,
    style,
    titleStyle,
    ...extraProps
  } = props;

  const onPressCallback = (event: GestureResponderEvent) => {
    if (isDisabled || isLoading) {
      return;
    }
    if (typeof onPress === 'function') {
      onPress(event);
    }
  };

  return (
    <Touchable
      style={[styles.container, style]}
      {...extraProps}
      onPress={onPress ? onPressCallback : undefined}
    >
      {title && (
        <Text
          style={[
            styles.title,
            strong && styles.strong,
            accent && styles.accent,
            titleStyle
          ]}
        >
          {title}
        </Text>
      )}
    </Touchable>
  );
}

export default ButtonLink;
