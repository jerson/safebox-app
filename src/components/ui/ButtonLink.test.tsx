import React from 'react';
import * as renderer from 'react-test-renderer';
import ButtonLink, {ButtonLinkProps} from './ButtonLink';

const props: ButtonLinkProps = {
  title: 'sample',
};
test('renders without crashing', () => {
  const snapshot = renderer.create(<ButtonLink {...props} />).toJSON();
  expect(snapshot).toBeTruthy();
});

test('render just views', () => {
  const snapshot = renderer.create(<ButtonLink {...props} />).toJSON();
  expect(snapshot).toMatchSnapshot();
});
