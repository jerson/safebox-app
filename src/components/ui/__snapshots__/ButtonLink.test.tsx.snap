// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render just views 1`] = `
<_class
  style={
    Array [
      Object {
        "alignSelf": "center",
        "padding": 6,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Nunito",
          "fontSize": undefined,
          "fontStyle": "normal",
          "fontWeight": "400",
        },
        Object {},
        Array [
          Object {
            "color": "#65748b",
            "fontSize": 13,
          },
          undefined,
          undefined,
          undefined,
        ],
      ]
    }
  >
    sample
  </Text>
</_class>
`;
